### ISSUE TITLE
WEBP_LEVEL_OVERVIEW doesn't affect the compression 

### ISSUE THREAD

**Issue by IdanAviv89** [OSGeo/gdal#6550]
Hi, 
I have a raster where I use gdaladdo to add overviews. When I use the parameter of WEBP_LEVEL_OVERVIEW=75, I get the result compression is lossless. 
I run the following command:
`gdaladdo --config GDAL_NUM_THREADS ALL_CPUS --config COMPRESS_OVERVIEW WEBP --config WEBP_LEVEL_OVERVIEW 75 output.tif 2 4 8 16 32`
And when I do gdalinfo (`gdalinfo output.tif -oo OVERVIEW_LEVEL=0`):
`Image Structure Metadata:
  COMPRESSION=WEBP
  COMPRESSION_REVERSIBILITY=LOSSLESS
  INTERLEAVE=PIXEL`

When I do overview lossless on lossy raster, I get that the COMPRESSION_REVERSIBILITY=LOSSY. 
I made another test, where I took a raster and duplicate it and add overview once 75 and the other time 100. I got that both file have the same size.
It's looks like the parameter of the quality doesn't affect the overview quality level, but it only being affected by the raster quality.

thanks in advance,
idan 

gdal version 3.6-dev

**Comment by rouault** [comment]
I dont reproduce this with latest master, doing the following:
```
$ gdal_translate autotest/gdrivers/data/small_world.tif out.tif  -co COMPRESS=WEBP
$ gdaladdo --config GDAL_NUM_THREADS ALL_CPUS --config COMPRESS_OVERVIEW WEBP --config WEBP_LEVEL_OVERVIEW 75 out.tif 2 4 8 16
$ ll out.tif
-rw-rw-r-- 1 even even 28308 oct.  19 16:52 out.tif
$ gdalinfo out.tif  -oo OVERVIEW_LEVEL=0
[...]
Image Structure Metadata:
  COMPRESSION=WEBP
  INTERLEAVE=PIXEL
[...]
$ gdal_translate autotest/gdrivers/data/small_world.tif out.tif  -co COMPRESS=WEBP
$ gdaladdo --config GDAL_NUM_THREADS ALL_CPUS --config COMPRESS_OVERVIEW WEBP --config WEBP_LEVEL_OVERVIEW 50 out.tif 2 4 8 16
$ ll out.tif
-rw-rw-r-- 1 even even 27058 oct.  19 16:53 out.tif
$ gdalinfo out.tif  -oo OVERVIEW_LEVEL=0
[...]
Image Structure Metadata:
  COMPRESSION=WEBP
  INTERLEAVE=PIXEL
[...]
```

As mentioned in another ticket, the COMPRESSION_REVERSIBILITY hint is only implemented for the main resolution, not overviews, so I don't see how you woulg get it.

**Comment by IdanAviv89** [comment]
hi, 
Thanks for the quick respond.

I am running the following test:

1. `gdal_translate small_world.tif output.tif -of GTIFF -co COMPRESS=WEBP -co TILED=YES -co BLOCKXSIZE=512 -co BLOCKYSIZE=512 -co WEBP_LEVEL=100 -co WEBP_LOSSLESS=True  -co NUM_THREADS=ALL_CPUS`

2. duplicate the output.tif to two files (output_overview_100.tif and output_overview_75.tif).

3.  `gdaladdo --config GDAL_NUM_THREADS ALL_CPUS --config COMPRESS_OVERVIEW WEBP --config WEBP_LEVEL_OVERVIEW 100 --config WEBP_LOSSLESS_OVERVIEW TRUE output_overview_100.tif 2 4 8 16 32`

4. `gdaladdo --config GDAL_NUM_THREADS ALL_CPUS --config COMPRESS_OVERVIEW WEBP --config WEBP_LEVEL_OVERVIEW 75 output_overview_75.tif 2 4 8 16 32`

I get that the two images are in the same size and I get that the checksum of the overviews is the same between the two.  

`gdalinfo output_overview_100.tif -checksum`
```
Band 1 Block=512x512 Type=Byte, ColorInterp=Red
  Checksum=30111
  Overviews: 200x100, 100x50, 50x25, 25x13, 13x7
  Overviews checksum: 6233, 50775, 12725, 3212, 844
Band 2 Block=512x512 Type=Byte, ColorInterp=Green
  Checksum=32302
  Overviews: 200x100, 100x50, 50x25, 25x13, 13x7
  Overviews checksum: 7706, 51063, 12539, 3315, 872
Band 3 Block=512x512 Type=Byte, ColorInterp=Blue
  Checksum=40026
  Overviews: 200x100, 100x50, 50x25, 25x13, 13x7
  Overviews checksum: 26085, 55262, 13553, 3505, 980
```


`gdalinfo output_overview_75.tif -checksum `

```
Band 1 Block=512x512 Type=Byte, ColorInterp=Red
  Checksum=30111
  Overviews: 200x100, 100x50, 50x25, 25x13, 13x7
  Overviews checksum: 6233, 50775, 12725, 3212, 844
Band 2 Block=512x512 Type=Byte, ColorInterp=Green
  Checksum=32302
  Overviews: 200x100, 100x50, 50x25, 25x13, 13x7
  Overviews checksum: 7706, 51063, 12539, 3315, 872
Band 3 Block=512x512 Type=Byte, ColorInterp=Blue
  Checksum=40026
  Overviews: 200x100, 100x50, 50x25, 25x13, 13x7
  Overviews checksum: 26085, 55262, 13553, 3505, 980
```

In the other [ticket](https://github.com/OSGeo/gdal/pull/6442#issuecomment-1260890236) you said :

`Is there any way to do it also for the overviews ?`
```
yes:

$ gdal_translate byte.tif test.tif
$ gdaladdo test.tif 2 --config COMPRESS_OVERVIEW JXL
$ gdalinfo test.tif -oo OVERVIEW_LEVEL=0
[...}
Image Structure Metadata:
  COMPRESSION=JXL
  COMPRESSION_REVERSIBILITY=LOSSLESS (possibly)
  INTERLEAVE=BAND
$ python -c "from osgeo import gdal; ds = gdal.Open('test.tif'); print(ds.GetRasterBand(1).GetOverview(0).GetDataset().GetMetadata('IMAGE_STRUCTURE'))"
{'COMPRESSION': 'JXL', 'COMPRESSION_REVERSIBILITY': 'LOSSLESS (possibly)', 'INTERLEAVE': 'BAND'}
```
So, I thought you could get it also the lossless/lossy information of the compression for the overviews.

**Comment by rouault** [comment]
ah the issue is that the overviews will inherit the  WEBP_LOSSLESS=True property of the full resolution image. So you need to explicitly set --config WEBP_LOSSLESS_OVERVIEW NO.

Regarding getting the COMPRESSION_REVERSIBILITY for overviews, this was my mistake. My GDAL build didn't have the WEBP driver builtin, which is needed to get that info.

So all in all, I think everything is OK.

**Comment by IdanAviv89** [comment]
You are right, my bad (the webp_lossless works).
Thanks for the help.

**Comment by rouault** [comment]
actually https://github.com/OSGeo/gdal/pull/6552 should make things more user friendly

**Comment by IdanAviv89** [comment]
I get the same problem in the cog driver:
1. `gdal_translate la.tif output_cog1.tif -of COG -co COMPRESS=WEBP -co QUALITY=100 -co OVERVIEW_COMPRESS=WEBP -co OVERVIEW_QUALITY=75 -co NUM_THREADS=ALL_CPUS`
2. `gdal_translate la.tif output_cog2.tif -of COG -co COMPRESS=WEBP -co QUALITY=100 -co OVERVIEW_COMPRESS=WEBP -co OVERVIEW_QUALITY=100 -co NUM_THREADS=ALL_CPUS`

this two result in the exact same checksum for the overview (and lossless for both of them).

would  [6552](https://github.com/OSGeo/gdal/pull/6552) fix it ?

**Comment by rouault** [comment]
> would [6552](https://github.com/OSGeo/gdal/pull/6552) fix it ?

perhaps. But I've issued an explicit PR in https://github.com/OSGeo/gdal/pull/6553 to make it more obvious (and that one can be backported)

**Comment by IdanAviv89** [comment]
Hi, 
When I am using the cog driver I get the following warning:
`Warning 1: WEBP_LEVEL is specified, but WEBP_LOSSLESS=YES. WEBP_LEVEL will be ignored.`
Even though I am not using the WEBP_LOSSLESS option.

**Comment by jratike80** [comment]
Could you add the whole command so others could test simply by copy-paste?

**Comment by rouault** [comment]
> When I am using the cog driver I get the following warning:

fixed per 74a8d6fb6783c53c92b50f102ac2849547a84ebe
